Source/WebCore/page/csp/ContentSecurityPolicy.cpp

```diff
@@ -754,6 +754,20 @@ bool ContentSecurityPolicy::allowBaseURI(const URL& url, bool overrideContentSec
     return allPoliciesAllow(WTFMove(handleViolatedDirective), &ContentSecurityPolicyDirectiveList::violatedDirectiveForBaseURI, url);
 }
 
+bool ContentSecurityPolicy::allowTrustedTypesPolicy(const String& value, bool isDuplicate, AllowTrustedTypePolicyDetails& details) const
```

Comment:
I know this is the current style but it is not great to have both bool and the details our parameter.
I doubt we can have return value being false and details being Allowed.

Could we just have it as:
```suggestion
AllowTrustedTypePolicyDetails ContentSecurityPolicy::computeTrustedTypesPolicy(const String& value, bool isDuplicate) const
```

Then in call site, we could do a switch.

I would be tempted to rename AllowTrustedTypePolicyDetails to AllowTrustedTypePolicy

Reply:
We can't have false and Allowed but we can have true and Disallowed because of report only mode. Though we only need access to details in this case within CSP itself not within the factory so I can probably refactor this like you suggest. Will give it a go.